moveit_experimental/moveit_jog_arm/src/jog_calcs.cpp

```diff
@@ -274,6 +274,16 @@ bool JogCalcs::cartesianJogCalcs(geometry_msgs::TwistStamped& cmd, JogArmShared&
     Eigen::Vector3d translation_vector(cmd.twist.linear.x, cmd.twist.linear.y, cmd.twist.linear.z);
     Eigen::Vector3d angular_vector(cmd.twist.angular.x, cmd.twist.angular.y, cmd.twist.angular.z);
 
+    // Set uncontrolled dimensions to 0 in command frame
+    mutex.lock();
```

Comment:
This function is part of the realtime loop of control systems we are working on. I think ``std::atomic`` would be more appropriate here to prevent locking. 
https://en.cppreference.com/w/cpp/atomic/atomic

@tylerjw what do you think?

Reply:
IMO that's beyond the scope of what Adam should change in this PR (but probably a good idea to do later).

Reply:
It seems we're already moving towards atomic, though:
https://github.com/ros-planning/moveit/commit/bc752b7c8aef3826df3409efad3756c832ccad7e

Reply:
I looked at the above commit and moved to the std::array<std::atomic<bool>, 6> for the `control_dimensions`. Tested and working on our system